parquet-thrift/src/main/java/parquet/thrift/ThriftSchemaConverter.java

```diff
@@ -82,6 +85,37 @@ private void checkUnmatchedProjectionFilter(FieldProjectionFilter filter) {
     return new ThriftStructConverter().toStructType(thriftClass);
   }
 
+  /**
+   * Returns whether the given type is the element type of a list or is a
+   * synthetic group with one field that is the element type. This is
+   * determined by checking whether the type can be a synthetic group and by
+   * checking whether a potential synthetic group matches the expected
+   * ThriftField.
+   * <p>
+   * This method never guesses because the expected ThriftField is known.
+   *
+   * @param repeatedType a type that may be the element type
+   * @param thriftElement the expected Schema for list elements
+   * @return {@code true} if the repeatedType is the element schema
+   */
+  static boolean isElementType(Type repeatedType, ThriftField thriftElement) {
+    if (repeatedType.isPrimitive() ||
```

Comment:
There's no need for this check because we know the expected type, `thriftElement`. This method determines whether the thrift structure matches the repeated type or if the repeated type must contain an element that matches the thrift structure. Matching by the structure itself avoids making decisions based on field names.